integrations/aws/main.py

```diff
@@ -0,0 +1,168 @@
+from port_ocean.context.ocean import ocean
+from loguru import logger
+from starlette.requests import Request
+from port_ocean.core.ocean_types import ASYNC_GENERATOR_RESYNC_TYPE
+
+@ocean.on_resync()
+async def resync_all(kind: str) -> ASYNC_GENERATOR_RESYNC_TYPE:
+    if kind in iter(ResourceKindsWithSpecialHandling):
+        logger.info("Kind already has a specific handling, skipping", kind=kind)
+        return
+    async for batch in resync_cloudcontrol(kind):
+        yield batch
+
+@ocean.on_resync(kind=ResourceKindsWithSpecialHandling.CLOUDRESOURCE)
+async def resync_generic_cloud_resource(kind: str) -> ASYNC_GENERATOR_RESYNC_TYPE:
+    resource_kinds = get_resource_kinds_from_config(kind)
+
+    if len(resource_kinds) == 0:
+        logger.error(f"Resource kinds not found in port app config for {kind}")
+        return
+
+    for kind in resource_kinds:
+        async for batch in resync_cloudcontrol(kind):
+            yield batch
+
+
+@ocean.on_resync(kind=ResourceKindsWithSpecialHandling.ACM)
+async def resync_acm(kind: str) -> ASYNC_GENERATOR_RESYNC_TYPE:
+    sessions = _get_sessions()
+    for batch in describe_resources(sessions, 'acm', 'list_certificates', 'CertificateSummaryList', 'NextToken'):
+        yield batch
+
+@ocean.on_resync(kind=ResourceKindsWithSpecialHandling.LOADBALANCER)
+async def resync_loadbalancer(kind: str) -> ASYNC_GENERATOR_RESYNC_TYPE:
+    sessions = _get_sessions()
+    for batch in describe_resources(sessions, 'elbv2', 'describe_load_balancers', 'LoadBalancers', 'NextMarker'):
+        yield batch
+
+@ocean.on_resync(kind=ResourceKindsWithSpecialHandling.CLOUDFORMATION)
+async def resync_cloudformation(kind: str) -> ASYNC_GENERATOR_RESYNC_TYPE:
+    sessions = _get_sessions()
+    for batch in describe_resources(sessions, 'cloudformation', 'list_stacks', 'StackSummaries', 'NextToken'):
+        yield batch
+
+
+async def resync_cloudcontrol(kind: str) -> ASYNC_GENERATOR_RESYNC_TYPE:
+    sessions = _get_sessions()
+    next_token = None
+    for session in sessions:
+        region = session.region_name
+        account_id = find_account_id_by_session(session)
+        while True:
+            all_instances = []
+            try:
+                cloudcontrol = session.client('cloudcontrol')
+                params = {
+                    'TypeName': kind,
+                }
+                if next_token:
+                    params['NextToken'] = next_token
+                
+                response = cloudcontrol.list_resources(**params)
+                next_token = response.get('NextToken')
+                for instance in response.get('ResourceDescriptions', []):
+                    described = describe_single_resource(kind, instance.get('Identifier'), account_id, region)
+                    described.update({'Kind': kind, ACCOUNT_ID_PROPERTY: account_id})
+                    all_instances.append(_fix_unserializable_date_properties(described))
```

Comment:
why is that for? this means that for every resource that we query as part of the list, we also perform another specific query for it? I thought you told me that you were able to overcome this.

Reply:
Yeah, when listing resources using `cloudcontrol` alone some of the properties do not arrive,
for example:
  - `Arn` do not exist when simply listing `cloudformation` resources using cloudcontrol